src/io.cpp

```diff
@@ -0,0 +1,61 @@
+#include <sstream>
+#include "io.h"
+#include <memory>
+
+using namespace SumColouring;
+
+std::unique_ptr<std::pair<int, int>> parseLine(const std::string &line) {
+    std::stringstream ss(line);
+
+    std::string temp;
+    ss >> temp;
+    int firstColumn = strtol(temp.c_str(), nullptr, 10);
+    ss >> temp;
+    int secondColumn = strtol(temp.c_str(), nullptr, 10);
+
+    return std::unique_ptr<std::pair<int, int>>(new std::pair<int, int>{firstColumn, secondColumn});
+}
+
+std::unique_ptr<std::pair<int, int>> readNextLine(std::istream &in) {
```

Comment:
You could just use >> operator, e.g.
int a, b;
in >> a >> b;
it will read two integers from the stream, separated by white characters